libbpfgo.go

```diff
@@ -1031,7 +1031,7 @@ func (it *BPFMapIterator) Next() bool {
 	nextPtr := unsafe.Pointer(&next[0])
 
 	errC, err := C.bpf_map_get_next_key(it.b.fd, prevPtr, nextPtr)
-	if errno, ok := err.(syscall.Errno); errC == -1 && ok && errno == C.ENOENT {
+	if errno, ok := err.(syscall.Errno); errC == -2 && ok && errno == C.ENOENT {
```

Comment:
Why is this?

Reply:
That's the return code for the function call, one of the map tests failed because of this change.

Reply:
Got it! I see now. i guess it's redundant to check `errno == C.ENOENT` and `errC = -2`. Do you think we should simplify to just the following?: 
```suggestion
	if errC == -2 {
```

Reply:
This is actually outside the scope of this PR, so I'd say leave it for now and we can reassess before the next release